driver/src/models/standing_order.rs

```diff
@@ -28,10 +28,10 @@ impl ConcatenatingHashable for Vec<StandingOrder> {
         for i in 0..models::NUM_RESERVED_ACCOUNTS {
             hasher.input(self
                 .iter()
-                .position(|o| o.account_id == i as u16) 
+                .position(|o| o.account_id == u16::from(i))
```

Comment:
This one is a really good find. Turns out that using as u16 would also work if i was e.g. a u128 (and then just cut off 112 bits)

Reply:
Yes, from what the error message was saying, this was unsafe because it would truncate the number.